src/test/java/seedu/address/logic/commands/SetGoalCommandTest.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showPersonAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalMeetings.MEETING_ONE;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import java.util.Locale;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Goal;
+import seedu.address.model.person.Meeting;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+
+class SetGoalCommandTest {
+
+    private static final Meeting VALID_MEETING = MEETING_ONE;
+
+    private final Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_setGoalUnfilteredList_success() {
+        Person firstPerson = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+        Person editedPerson = new PersonBuilder(firstPerson).withGoal(new Goal(Goal.Frequency.WEEKLY)).build();
+
+        SetGoalCommand cmd = new SetGoalCommand(INDEX_FIRST_PERSON, Goal.Frequency.WEEKLY);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+        expectedModel.setPerson(firstPerson, editedPerson);
+
+        String expectedMessage = String.format(
+                SetGoalCommand.MESSAGE_ADD_GOAL_SUCCESS,
+                editedPerson.getGoal().toString().toLowerCase(Locale.ROOT),
```

Comment:
You should create another method on `Goal` class that abstracts this out. Maybe name it `toUi()` or something